front/src/components/CodeEditorWindow.js

```diff
@@ -0,0 +1,23 @@
+import React, {useState} from 'react';
+
+import Editor from '@monaco-editor/react';
+
+const CodeEditorWindow = ({onChange, code}) => {
+  const [value, setValue] = useState(code || '');
+
+  const handleEditorChange = (value) => {
+    setValue(value);
+    onChange('code', value);
+  };
```

Comment:
weird way of doing `handleEditorChange`. why not passing in argument the event and get the value from it ? Here need to call onChange a second time.

Reply:
It was the way it was done in a tutorial. Maybe I could use only setCode, and not use the value state. But maybe we could modify it later?

Reply:
yup